core/src/replay_stage.rs

```diff
@@ -1469,11 +1462,104 @@ impl ReplayStage {
             vote_signatures.clear();
         }
 
-        let _ = cluster_info.send_vote(
-            &vote_tx,
-            crate::banking_stage::next_leader_tpu(cluster_info, poh_recorder),
+        Some(vote_tx)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn refresh_last_vote(
+        tower: &mut Tower,
+        cluster_info: &ClusterInfo,
+        heaviest_bank_on_same_fork: &Bank,
+        poh_recorder: &Mutex<PohRecorder>,
+        my_latest_landed_vote: Slot,
+        vote_account_pubkey: &Pubkey,
+        authorized_voter_keypairs: &[Arc<Keypair>],
+        vote_signatures: &mut Vec<Signature>,
+        has_new_vote_been_rooted: bool,
+        last_vote_refresh_time: &mut Instant,
+    ) {
+        let last_voted_slot = tower.last_voted_slot();
+        if last_voted_slot.is_none() {
+            return;
+        }
+
+        // Refresh the vote if our latest vote hasn't landed, and the recent blockhash of the
+        // last attempt at a vote transaction has expired
+        let last_voted_slot = last_voted_slot.unwrap();
+        if my_latest_landed_vote >= last_voted_slot
+            || heaviest_bank_on_same_fork
+                .check_hash_age(&tower.last_vote_tx_blockhash(), MAX_PROCESSING_AGE)
+                .unwrap_or(false)
+            // In order to avoid voting on multiple forks all past MAX_PROCESSING_AGE that don't
+            // include the last voted blockhash
+            || last_vote_refresh_time.elapsed().as_millis() < MAX_VOTE_REFRESH_INTERVAL_MILLIS as u128
```

Comment:
I don't think this is needed because you call `refresh_last_vote_tx_blockhash` whenever a vote is refreshed

Reply:
Yeah, but the issue is you can hop between multiple forks, if they're all descended from your last vote. None of the forks' blockhashes will be included in the others, so checking using the latest refreshed blockhash may not be enough

Reply:
Ah got it, sounds good!